kernel/Dockerfile.bcc

```diff
@@ -0,0 +1,105 @@
+  luajit-dev \
+  m4 \
+  python \
+  zlib-dev \
+  && true
+
+RUN ln -s /usr/lib/cmake/llvm5/ /usr/lib/cmake/llvm; \
+    ln -s /usr/include/llvm5/llvm-c/ /usr/include/llvm-c; \
+    ln -s /usr/include/llvm5/llvm/ /usr/include/llvm
+
+WORKDIR /build
+COPY ./bcc.patches/ ./
+COPY Dockerfile.bcc /
+COPY ./bcc.patches/error.h /usr/include/
+COPY ./bcc.patches/cdefs.h /usr/include/sys/
+ENV ELFUTILS_VERSION=0.165
+ENV ELFUTILS_SHA256="a7fc9277192caaa5f30b47e8c0518dbcfd8c4a19c6493a63d511d804290ce972"
+RUN curl -sSL -O https://fedorahosted.org/releases/e/l/elfutils/0.165/elfutils-$ELFUTILS_VERSION.tar.bz2
+ 
+RUN echo "${ELFUTILS_SHA256}  /build/elfutils-$ELFUTILS_VERSION.tar.bz2" | sha256sum -c - && \
+  cat elfutils-$ELFUTILS_VERSION.tar.bz2 | tar xjf - && \
+  cd elfutils-$ELFUTILS_VERSION && \
+  patch -p1 < ../100-musl-compat.patch && \
+  patch -p0 < ../decl.patch && \
+  patch -p0 < ../intl.patch
+ENV BCC_COMMIT=6972806729da00ecda0235abac61d66c8fad7fad
+RUN git clone https://github.com/iovisor/bcc.git && cd bcc && git checkout $BCC_COMMIT
+RUN cd bcc && patch -p0 < ../bcc-gnuism.patch && patch -p0 < ../bcc-lua.patch
+ENV LJSYSCALL_COMMIT=e587f8c55aad3955dddab3a4fa6c1968037b5c6e
+RUN git clone https://github.com/justincormack/ljsyscall.git && cd ljsyscall && git checkout $LJSYSCALL_COMMIT
+
+COPY --from=ksrc /kernel-headers.tar /build
+RUN tar xf /build/kernel-headers.tar
+
+COPY --from=ksrc /kernel-dev.tar /build
+RUN tar xf /build/kernel-dev.tar
+
+COPY --from=ksrc /kernel.tar /build
+RUN tar xf /build/kernel.tar
+
+RUN cd /build
+RUN cd elfutils-$ELFUTILS_VERSION && \
+  aclocal && \
+  automake && \
+  ./configure --prefix=/usr CFLAGS=-Wno-strict-aliasing && \
+  make -C libelf && make -C libelf install
+RUN mkdir -p bcc/build && cd bcc/build && \
+  cmake .. -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_C_FLAGS="-I/build/usr/include" -DCMAKE_CXX_FLAGS="-I/build/usr/include" -DCMAKE_INSTALL_PREFIX=/usr -DLUAJIT_INCLUDE_DIR=/usr/include/luajit-2.1 && \
+  make && \
+  make install
+RUN mkdir -p /usr/local/share/lua/5.1/ && cd ljsyscall && cp -a *.lua syscall /usr/local/share/lua/5.1/
+
+ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/lib64
+
+FROM alpine:3.7 as mirror
+RUN mkdir -p /out/etc/apk && cp -r /etc/apk/* /out/etc/apk/
+RUN apk update && apk upgrade -a && \
+  apk add --no-cache --initdb -p /out \
+  busybox \
+  luajit \
+  python \
+  zlib \
+  && true
+
+FROM scratch
+ENTRYPOINT []
+CMD []
+WORKDIR /
+COPY --from=mirror /out /
+COPY --from=build /build/usr/src /usr/src/
+COPY --from=build /build/usr/include /usr/include/
+COPY --from=build /usr/lib/libelf* /usr/lib/
+COPY --from=build /usr/lib/libstdc* /usr/lib/
+COPY --from=build /usr/lib/libintl* /usr/lib/
+COPY --from=build /usr/lib64 /usr/lib/
+COPY --from=build /usr/lib/python2.7/site-packages /usr/lib/python2.7/site-packages/
+COPY --from=build /usr/share/bcc /usr/share/bcc/
+COPY --from=build /usr/bin/bcc-lua /usr/bin/
+COPY --from=build /usr/local/share/lua /usr/local/share/lua/
```

Comment:
These many `COPY` commands create many layers in the final image. It might be better to build the toot file system in `/out` in the previous stage and then have a single `COPY` statement here.